### PR TITLE
Reorganizing structure of middleware, updates to ctxlogger

### PR DESCRIPTION
**Pull request by abiduke612**
Working protovalidate/ctxlog, withCtx function for ctxlogger in middleware now

### CODE REVIEW

File: ctxlogger/CtxLogger.cs [hunk 1]
```diff
@@ -19,21 +28,161 @@ public CtxLoggerInterceptor(Serilog.ILogger logger)
...
+            foreach (var kvp in dictionary)
+            {
+                logger = logger.ForContext(kvp.Key, kvp.Value, destructureObjects: true);
+            }
+        }
+
+        // Add caller information to the log context
+        var location = new
+        {
+            function = callerMemberName,
```
Comment (toma3233) [inline ctxlogger/CtxLogger.cs:188]: How are we able to access this filename, line number, etc of where the log came from in the middleware? Add any useful links you may have used for this

File: ctxlogger/CtxLogger.cs [hunk 2]
```diff
@@ -19,21 +28,161 @@ public CtxLoggerInterceptor(Serilog.ILogger logger)
...
+
+public static class LoggerExtensions
+{
+    public static ILogger WithCtx(this ILogger logger, ServerCallContext context,
+        [CallerFilePath] string callerFilePath = "",
+        [CallerLineNumber] int callerLineNumber = 0,
+        [CallerMemberName] string callerMemberName = "")
+    {
+        // Extract JSON string from request headers
+        var json = context.RequestHeaders.GetValue("ctxlog-data");
```
Comment (toma3233) [inline ctxlogger/CtxLogger.cs]: please utilize a better variable name here instead of "json"

File: ctxlogger/CtxLogger.cs [hunk 3]
```diff
@@ -19,21 +28,161 @@ public CtxLoggerInterceptor(Serilog.ILogger logger)
...
+        {
+            var req = FilterLogs(message);
+            var ctxDictionary = new Dictionary<string, object>
+            {
+                { Constants.MethodFieldKey, context.Method },
+                { "request", req },
+                { "source", "CtxLog"}
+            };
+            // Serialize the dictionary to a JSON string
+            var json = JsonConvert.SerializeObject(ctxDictionary, new JsonSerializerSettings{TypeNameHandling = TypeNameHandling.Auto});
```
Comment (toma3233) [inline ctxlogger/CtxLogger.cs]: please use a more specific variable name rather than "json"

File: ctxlogger/CtxLogger.cs (hunk 1 above)
Comment (abiduke612) [reply ctxlogger/CtxLogger.cs:188]: added a link about Caller Information Attributes

File: policy/Policy.cs [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+                MaxRetries = 5
+            }
+        };
+
+        clientOptions.AddPolicy(new LoggingPolicy(logger), HttpPipelinePosition.PerCall);
+
+        return clientOptions;
+    }
+
+    public static StatusCode ConvertHTTPStatusToGRPCError(int httpStatusCode)
```
Comment (toma3233) [inline policy/Policy.cs:92]: please ensure to leave a comment here with the link that explains the reasoning for the conversion (same as the go middleware)

**Pull request closed without merging**